### PR TITLE
perf(lifecycle): add benchmark for lifecycle

### PR DESCRIPTION
**Pull request by hazzadous**
## Changes

Adds a basic lifecycle query. Also some notes on running benchmarks.

## How did you test this code?

added performance label to PR

Refers to https://github.com/PostHog/posthog/issues/7382

### CODE REVIEW

File: ee/benchmarks/README.md [hunk 1]
```diff
@@ -42,6 +42,14 @@ asv machine --machine ci-benchmarks --config ee/benchmarks/asv.conf.json
 CLICKHOUSE_HOST=X CLICKHOUSE_USER=X CLICKHOUSE_PASSWORD=X CLICKHOUSE_DATABASE=posthog asv run --config ee/benchmarks/asv.conf.json
 ```
 
+You'll probably want to be running one test, with quick iteration. Running e.g.:
+
+```
+asv run --config ee/benchmarks/asv.conf.json --bench track_lifecycle --quick
```
Comment (macobo) [inline ee/benchmarks/README.md:48]: I've also used --environment existing to skip the virtualenv setup

**Comment by macobo** [comment]
Don't forget to manually start the workflow on master if you end up doing a PR before benchmark data gets refreshed

**Review by macobo — APPROVED** [review]
We should merge this - nothing to draft really :)

File: ee/benchmarks/benchmarks.py [hunk 2]
```diff
@@ -458,6 +459,26 @@ def track_retention_filter_by_person_property_materialized(self):
...
+            data={
+                "insight": "LIFECYCLE",
+                "events": [{"id": "$pageview", "name": "$pageview", "type": "events", "order": 0, "math": "total"}],
+                "display": "ActionsLineGraph",
+                "interval": "week",
+                "shown_as": "Lifecycle",
+                "date_from": "-14d",
+                "properties": [],
+                "compare": False,
+                "filter_test_accounts": True,
```
Comment (macobo) [inline ee/benchmarks/benchmarks.py:474]: Nit: you can remove this + compare + properties and you _should_ remove interval and date_from as they're overridden.

File: ee/benchmarks/benchmarks.py [hunk 3]
```diff
@@ -458,6 +459,26 @@ def track_retention_filter_by_person_property_materialized(self):
 
         ClickhouseRetention().run(filter, self.team)
 
+    @benchmark_clickhouse
+    def track_lifecycle(self):
+        filter = Filter(
+            data={
+                "insight": "LIFECYCLE",
+                "events": [{"id": "$pageview", "name": "$pageview", "type": "events", "order": 0, "math": "total"}],
```
Comment (macobo) [inline ee/benchmarks/benchmarks.py:467]: Remove math and order - these are not used in any way.

File: ee/benchmarks/README.md (hunk 1 above)
Comment (hazzadous) [reply ee/benchmarks/README.md:48]: Ah great I missed that one 😳

**Comment by hazzadous** [comment]
> We should merge this - nothing to draft really :)

Cool I'll get in and address the comments separately then 🎉

**Pull request merged**